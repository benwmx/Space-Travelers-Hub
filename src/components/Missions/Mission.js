import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { toggleMissionJoin } from '../../redux/missions/missions';
import Styles from './mission.module.scss';

const Mission = (props) => {
  const { missionInfo } = props;
  const dispatch = useDispatch();

  const toggleJoin = () => {
    dispatch(toggleMissionJoin(missionInfo.id));
  };

  return (
    <tr className={(missionInfo.joined) ? Styles.joined : undefined}>
      <td><p className={Styles.missionName}>{missionInfo.name}</p></td>
      <td className={Styles.description}>{missionInfo.description}</td>
      <td>
        {
          (missionInfo.joined)
            ? <p className={Styles.activeMember}>Active Member</p>
            : <p className={Styles.notMember}>Not A Member</p>
        }
      </td>
      <td>
        {
          (missionInfo.joined)
            ? <button type="button" className={Styles.leaveMission} onClick={toggleJoin}> Leave Mission </button>
            : <button type="button" className={Styles.joinMission} onClick={toggleJoin}> Join Mission </button>
        }
      </td>
    </tr>
  );
};

Mission.propTypes = {
  missionInfo: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    joined: PropTypes.bool.isRequired,
  }).isRequired,
};

export default Mission;
