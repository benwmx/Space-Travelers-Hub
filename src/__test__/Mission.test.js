import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Mission from '../components/Missions';

describe('Mission Component test', () => {
  it('Should render', () => {
    const app = renderer.create(
      <Provider store={store}>
        <Mission />
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});
