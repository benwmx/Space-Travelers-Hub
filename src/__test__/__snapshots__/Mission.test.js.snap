// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Component test Should render 1`] = `
<table
  className="missionsTable"
>
  <thead
    className="headTable"
  >
    <tr>
      <td>
        Mission
      </td>
      <td>
        Description
      </td>
      <td>
        Status
      </td>
      <td>
        Join
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <p
          className="missionName"
        >
          
        </p>
      </td>
      <td
        className="description"
      >
        
      </td>
      <td>
        <p
          className="notMember"
        >
          Not A Member
        </p>
      </td>
      <td>
        <button
          className="joinMission"
          onClick={[Function]}
          type="button"
        >
           Join Mission 
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
